src/util/aabb2.ts

```diff
@@ -1,24 +1,39 @@
-import { vec2 } from 'gl-matrix'
-
 import { Immutable } from '~/types/immutable'
 
-export type Aabb2 = [vec2, vec2]
+// x1, y1, x2, y2
+export type Aabb2 = [number, number, number, number]
+
+export enum Elem {
+  x1,
+  y1,
+  x2,
+  y2,
+}
```

Comment:
It might be better simply to do this:

```
export const x1 = 0
export const y1 = 1
...
```

However, this being an optimization around memory, I would want to use [`d8`](https://v8.dev/docs/d8) or something to confirm that the bytecode generator turns `const` numbers into assembly immediates, rather than doing a variable lookup.